lib/HistoryModelService.js

```diff
@@ -6,52 +6,62 @@ var WaterlineLifecycleCallback = require('./WaterlineLifecycleCallback');
 
 var self = {
   createModels: function createModels(sails) {
-    async.auto({
-      loadCollection: createAllModelHistory,
-      loadModel: ['loadCollection', loadModels]
-    }, function(err, result) {
-      if (err) {
-        sails.log.debug('Error creating history models ', err);
-      } else {
-        sails.log.info('Successful creating of history models');
-      }
-    });
+    try {
+      async.auto({
+        loadCollection: createAllModelHistory,
+        tearDown: ['loadCollection', tearDownDatabase],
+        loadModel: ['tearDown', loadModels]
+      }, function(err, result) {
+        if (err) {
+          sails.log.debug('Error creating history models ', err);
+        } else {
+          sails.log.info('Successful creating of history models');
+        }
+      });
+    }catch(e){
+      sails.log.debug('Error creating history models');
+    }
 
-    function createAllModelHistory(asyncCb, asyncRes) {
+    function createAllModelHistory(asyncCb) {
       if (!sails.models) {
-        asyncCb({ message: 'No models!' });
+        asyncCb({
+          message: 'No models!'
+        });
       }
       _.each(sails.models, function eachInstantiatedModel(model, identity) {
-        var fields = model.attributes;
         WaterlineLifecycleCallback.extendLifecycleCallbackMethod(model);
+
         model.associations = populateAssociations(model);
         var waterlineModel = Waterline.Collection.extend(model);
 
         orm.loadCollection(waterlineModel);
+        createHistory(model);
+        return asyncCb(null, true);
+      });
+    }
 
-        if (model.history) {
-          var historyFields = {};
-          var originalFields = _.clone(fields);
-          _.extend(historyFields, HistoryModelFields.attributes, model.attributes);
-          var historyTableName = model.tableName ? model.tableName : model.identity;
+    function createHistory(model) {
+      if (model.history) {
```

Comment:
If will wrap the whole function with if, better not to invoke the function if will not satisfy the condition.